### DISCUSSION TITLE
How to make application in podman run faster?

### DISCUSSION THREAD

**Discussion by BinaryWorld2099** [containers/podman#18084] [Q&A]
I am working on a project that trying to deploy a java application in podman container in production.
But the preformane is 20% to 30% drop when compare to traditional way installation
Are there any tunning making application in podman as fast as before no using podman?

**Comment by rhatdan** [comment]
What is the output of `podman info` are you using the storage backend of `vfs`.
There should be no difference in performance between containerized and not, unless something is set very wrong.

**Comment by rhatdan** [comment]
One other thing would be if you are using fuse-overlayfs by default?

**Reply by BinaryWorld2099** [reply]
seems JMS and network is slow in podman container
Any suggest to imporve it?

**Comment by BinaryWorld2099** [comment]
It is java application runs on java 1.8

below is the podman info output

root@master:~# podman info
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: 'conmon: /usr/bin/conmon'
    path: /usr/bin/conmon
    version: 'conmon version 2.0.25, commit: unknown'
  cpus: 6
  distribution:
    codename: jammy
    distribution: ubuntu
    version: "22.04"
  eventLogger: journald
  hostname: master
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.15.0-40-generic
  linkmode: dynamic
  logDriver: journald
  memFree: 3148328960
  memTotal: 8137781248
  ociRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: |-
      runc version 1.1.0-0ubuntu1
      spec: 1.0.2-dev
      go: go1.17.3
      libseccomp: 2.5.3
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: true
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.0.1
      commit: 6a7b16babc95b6a3056b33fb45b74a6f62262dd4
      libslirp: 4.6.1
  swapFree: 0
  swapTotal: 0
  uptime: 1m 9.14s
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries: {}
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.4.4
  Built: 0
  BuiltTime: Thu Jan  1 08:00:00 1970
  GitCommit: ""
  GoVersion: go1.17.3
  OsArch: linux/amd64
  Version: 3.4.4

**Comment by BinaryWorld2099** [comment]
this is another server with more power resources 
host:
  arch: amd64
  buildahVersion: 1.27.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.4-1.module+el8.7.0+20877+e0f9ac15.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.4, commit: c8ca5f742bed8ec1371fdcdd3e63a623cbb16773'
  cpuUtilization:
    idlePercent: 94.07
    systemPercent: 1.75
    userPercent: 4.18
  cpus: 8
  distribution:
    distribution: '"ol"'
    variant: server
    version: "8.5"
  eventLogger: file
  hostname: test-vm
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 6001
      size: 1
    - container_id: 1
      host_id: 231072
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 6001
      size: 1
    - container_id: 1
      host_id: 231072
      size: 65536
  kernel: 5.4.17-2136.305.5.3.el8uek.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 280416256
  memTotal: 33369350144
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.1.4-1.module+el8.7.0+20877+e0f9ac15.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.1.4
      spec: 1.0.2-dev
      go: go1.18.4
      libseccomp: 2.5.1
  os: linux
  remoteSocket:
    path: /run/user/6001/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-2.module+el8.7.0+20877+e0f9ac15.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.1
  swapFree: 0
  swapTotal: 0
  uptime: 0h 13m 43.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - container-registry.oracle.com
  - docker.io
store:
  configFile: /projects/dockerusr/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/dockerusr/.local/share/containers/storage
  graphRootAllocated: 20097007616
  graphRootUsed: 13692706816
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/6001/containers
  volumePath: /home/dockerusr/.local/share/containers/storage/volumes
version:
  APIVersion: 4.2.0
  Built: 1669122264
  BuiltTime: Tue Nov 22 21:04:24 2022
  GitCommit: ""
  GoVersion: go1.18.4
  Os: linux
  OsArch: linux/amd64
  Version: 4.2.0
